packages/nimble-components/src/tabs/specs/scrollable-tabs.md

```diff
@@ -0,0 +1,106 @@
+
+#### Scroll buttons
+
+While there are no explicit ARIA guidelines for how dynamic buttons in something like a tablist should respond to the keyboard, the [interaction pattern for the carousel](https://www.w3.org/WAI/ARIA/apg/patterns/carousel/#keyboardinteraction) is fairly similar to this use-case, which suggests that the scroll buttons could be valid tab stops.
+
+However, as described in the ['Fundamental Keyboard Navigation Conventions'](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#fundamentalkeyboardnavigationconventions) section, it states:
+
+_"A primary keyboard navigation convention common across all platforms is that the tab and shift + tab keys move focus from one UI component to another while other keys, primarily the arrow keys, move focus inside of components that include multiple focusable elements"_.
+
+The above statement would suggest that we might implement a roving tab index for the focusable elements within the `Tabs` component, whereby a user would use the arrow keys (once some element within the `Tabs` component has focus via the `Tab` key) to move between the interactive elements including the scroll buttons.
+
+**Carousel strategy**:
+
+Pros:
+
+-   Easiest to implement (i.e. it's free as it behaves this way out of the box)
+-   Not too cumbersome as one shouldn't expect a particular page to have a multitude of `Tabs` components, and additionally, the extra tabs are only if the buttons are even visible.
+
+Cons:
+
+-   After interacting with a scroll button via the keyboard a user would be required to use the mouse to select a tab that is currently in view (if a completely new set of tabs was scrolled into view).
+-   Possibly goes against conventions for composite components
+-   Possibly not in-line with other Nimble conventions such as not having the inc/dec buttons on a numeric behave as tab stops.
+
+Related:
+
+-   The [Shoelace design system](https://shoelace.style/components/tab-group#scrolling-tabs) has an example of this approach
+
+**Roving tab strategy**:
+
+Pros:
+
+-   Seems like it might be more in-line with ARIA conventions
+-   Scrolling behavior can still be achievable through other keyboard interactions like `PageUp/PageDown`, which ARIA doesn't provide any explicit guidance for or against.
+-   Tabbing through the page may seem nice without the two extra stops that may be present with the scroll buttons.
+
+Cons:
+
+-   More expensive to implement
+-   Getting to the scroll buttons could now be overly cumbersome when tab focus is on a tab.
+
+**Scroll buttons are unreachable via keyboard**:
+
+Pros:
+
+-   Keyboard-only users won't run into issues with interactions with scroll buttons leading them down an unexpected path (i.e. pressing arrow keys would undo their scrolling).
```

Comment:
> i.e. pressing arrow keys would undo their scrolling

Can you clarify? I didn't understand this.

Reply:
I'm trying to speak to the point that if the scroll buttons were focusable and a user attempted to use the keyboard to perform a scroll through them, then they might be under the impression that pressing the arrow key would move them to the first visible tab button in the view, but I was asserting that it would not, and instead move the focus one to the right/left of the currently focused tab (which would then unroll all of the scrolling they might have done via the scroll buttons).

That said, I suppose we _could_ consider the idea that if a user did use the scroll buttons via the keyboard that pressing the arrow key would _either_ move them to the right/left of the currently focused tab _if_ it was still in view, _or_ to the first visible tab. However, this seems like an overly complicated implementation which doesn't even seem like that much of an improvement over just not making the scroll buttons accessible via the keyboard.